core/pkg/service/flag-evaluation/flag_evaluator_v2.go

```diff
@@ -0,0 +1,274 @@
+package service
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	evalV1 "buf.build/gen/go/open-feature/flagd/protocolbuffers/go/flagd/evaluation/v1"
+	"connectrpc.com/connect"
+	"github.com/open-feature/flagd/core/pkg/evaluator"
+	"github.com/open-feature/flagd/core/pkg/logger"
+	"github.com/open-feature/flagd/core/pkg/service"
+	"github.com/open-feature/flagd/core/pkg/telemetry"
+	"github.com/rs/xid"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/trace"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+type FlagEvaluationServiceV2 struct {
+	logger                *logger.Logger
+	eval                  evaluator.IEvaluator
+	metrics               *telemetry.MetricsRecorder
+	eventingConfiguration *eventingConfiguration
+	flagEvalTracer        trace.Tracer
+}
+
+// NewFlagEvaluationServiceV2 creates a FlagEvaluationService with provided parameters
+func NewFlagEvaluationServiceV2(log *logger.Logger,
```

Comment:
This is a nit, but I think it might be better not postfix this with `V2`. I understand this is just a namespacing, but I think we risk some confusion if we ever actually use `flagd.evaluation.v2` (right now this is associated with `flagd.evaluation.v1`.

Reply:
Or maybe rename the old evaluation service to something else.

Reply:
> Or maybe rename the old evaluation service to something else.

Actually that might be better.